src/components/NowPlaying.tsx

```diff
@@ -41,14 +52,49 @@ class NowPlaying extends React.Component<{}, {audioFeatures: AudioFeaturesRespon
     apiCall();
   }
   
-  playIcon = <img className={styles.playIcon} src={"https://img.icons8.com/?size=100&id=36067&format=png&color=FFFFFF"}/>;
+  generateRecommendations = async () => {
+    let apiOptions = new GetRecommendationsInput();
+    apiOptions.data.seed_tracks = this.state.queue.toString();
+    var recommendations = await getRecommendations(apiOptions);
+    this.setState({
+      recommendations: recommendations,
+    });
+  };
+
+  addToQueue = (event?: Event & {data: number}) => {
+    if (!event || !Spicetify.Player.data || this.state.queue.includes(Spicetify.Player.data.item.uri.split(":")[2])) {
+      return;
+    }
+
+    let progressPercentage = (event.data / Spicetify.Player.data.item.duration.milliseconds) * 100;
+    if (progressPercentage < 50) {
+      return;
+    }
 
-  nowPlayingStyles = {
-    primaryTrackInfo: {}
+    let newQueue = this.state.queue.slice();
+    if (this.state.queue.length == 5) {
+      newQueue.shift();
+    }
+
+    newQueue.push(Spicetify.Player.data.item.uri.split(":")[2]);
+    this.setState({
+      queue: newQueue,
+    }, () => {
+      if (Spicetify.LocalStorage.get("queue") == this.state.queue.toString()) {
+        return;
+      }
+      Spicetify.LocalStorage.set("queue", this.state.queue.toString());
+      this.generateRecommendations();
+    });
   }
```

Comment:
Also would it be possible to place the code pertaining to generating recommendations within the `DynamicRecommendations` component file? We could just change it so that all of our logic is within `NowPlaying`, but I think it would be less clean and also you would have to transfer everything that's currently in there to this file.